lib/algora/terminate.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Algora.Terminate do
+  use GenServer
+
+  import Ecto.Query, warn: false
+
+  @terminate_interval :timer.se(10)
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+  end
+
+  @impl true
+  def init(state) do
+    schedule_terminate()
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_info(:terminate, state) do
+    from(v in Algora.Library.Video,
+          where: v.duration == 0 and v.is_live == false,
+          select: v.id
+        )
+        |> Algora.Repo.Local.all
+        |> Enum.each(&Algora.Library.terminate_stream/1)
```

Comment:
Let's move this chunk into a function like `Library.terminate_interrupted_streams()`